system/lib/wasmfs/wasmfs.h

```diff
@@ -36,10 +37,12 @@ class WasmFS {
   void preloadFiles();
 
 public:
+  std::vector<std::unique_ptr<Backend>> backendTable;
```

Comment:
Is this meant to be public? I'd expect it to be private, and to have methods to get a backend by its index (assuming we are going with the handle approach and not raw pointers).

Reply:
Hmmm, I was considering sticking with the raw pointer approach.

Reply:
Raw pointers sgtm. Separately, though, this being public doesn't seem optimal?

Reply:
Agreed, I am going to add a lock around it as well.